dashboard/app/controllers/courses_controller.rb

```diff
@@ -189,6 +143,53 @@ def i18n_params
 
   private
 
+  def get_unit_group
+    course_name = params[:course_name]
+
+    unit_group =
+      params[:action] == "edit" ?
+        UnitGroup.get_without_cache(course_name) :
+        UnitGroup.get_from_cache(course_name)
+
+    return unit_group if unit_group
+
+    # When the url of a course family is requested, redirect to a specific course version.
+    if UnitGroup.family_names.include?(params[:course_name])
+      unit_group = UnitGroup.latest_stable_version(params[:course_name])
+      redirect_to action: params[:action], course_name: unit_group.name if unit_group
```

Comment:
The check for `if unit_group` is new here. This makes sure we don't get Honeybadger errors for someone trying to go to a course family name that has no stable unit groups. [Example Error](https://codedotorg.slack.com/archives/C55JZ1BPZ/p1643052013005200)

Reply:
checking my understanding -- is the rest of this logic (besides `if unit_group`) moving here from somewhere else?

Reply:
Its an updated version of this removed code that used to start on line 74 
```
    elsif UnitGroup.family_names.include?(params[:course_name])
      # csp and csd are each "course families", each containing multiple "course versions".
      # When the url of a course family is requested, redirect to a specific course version.
      redirect_query_string = request.query_string.empty? ? '' : "?#{request.query_string}"
      redirect_to_course = UnitGroup.latest_stable(params[:course_name])
      redirect_to "#{course_path(redirect_to_course)}#{redirect_query_string}"
```

as well as these lines from standards action

```
    if !@unit_group.present? && UnitGroup.family_names.include?(params[:course_name])
      redirect_to_course = UnitGroup.latest_stable(params[:course_name])
      redirect_to standards_course_path(redirect_to_course)
      return
    end
```

I thought it made it more consistent but if you feel like we should keep it locked down to only show and standards I could update this

Reply:
I will say things like this also match the way scripts_controller is set up so that the two are more similar

Reply:
I guess you didn't believe me when I said "checking my understanding" 😂  no worries, Dani, this looks fine